app/auth/user_management.py

```
@@ -0,0 +1,80 @@
+            "last_name": custom_claims["last_name"],
+            "is_active": not user_record.disabled,
+            "created_at": str(user_record.user_metadata.creation_timestamp)
+        }
+    except Exception as e:
+        raise HTTPException(status_code=status.HTTP_500_INTERNAL_SERVER_ERROR, detail=str(e))
+
+
+@router.put("/profile")
+async def update_user_profile(
+    update_data: UserUpdateRequest,
+    current_user: Dict[str, Any] = Depends(get_current_user)
```

Ruff (0.14.8)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)